staging/src/k8s.io/kube-aggregator/pkg/controllers/status/available_controller.go

```diff
@@ -387,8 +387,10 @@ func (c *AvailableConditionController) sync(key string) error {
 						return
 					}
 
-					// setting the system-masters identity ensures that we will always have access rights
-					transport.SetAuthProxyHeaders(newReq, "system:kube-aggregator", []string{"system:masters"}, nil)
+					// make an authenticated request so that this is authorized via the system:discovery
+					// cluster role binding in environments that enable RBAC and authorization delegation
+					// avoid using the system:masters identity to prevent any possibility of SSRF
+					transport.SetAuthProxyHeaders(newReq, "system:kube-aggregator", nil, nil)
```

Comment:
should we at least be setting `system:authenticated`?

Reply:
I'm trying to think about the new failure modes this introduces... the main one I can think of is that by not using the privileged group, this will likely start requiring aggregated servers to do delegated authz checks, which could fail

what identity do we use for fetching openapi for aggregation?

Reply:
> should we at least be setting `system:authenticated`?

This does assume that the `NewAuthenticatedGroupAdder` code is in use.  That might be a bad assumption.

> I'm trying to think about the new failure modes this introduces... the main one I can think of is that by not using the privileged group, this will likely start requiring aggregated servers to do delegated authz checks, which could fail
> 
> what identity do we use for fetching openapi for aggregation?

https://github.com/kubernetes/kubernetes/blob/f9afd68e3b3fc05f3607ea7cb90808c3e931ba9c/staging/src/k8s.io/kube-aggregator/pkg/controllers/openapiv3/aggregator/downloader.go#L54

`system:aggregator` with no groups.  IIUC we do not stop routing traffic to an aggregated API if the open API stuff fails to download so this might not be a good comparison.